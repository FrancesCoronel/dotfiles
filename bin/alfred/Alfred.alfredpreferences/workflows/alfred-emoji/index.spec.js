jest.mock('alfy');

const alfy = require('alfy');
const emoji = require('./index');

it('should return empty array', async () => {
  alfy.__setResponse();
  await expect(emoji()).resolves.toMatchSnapshot();
});

it('should return return correct smiles for "rofl"', async () => {
  alfy.__setResponse({
    results: [
      { text: '😂', score: 0.42186877131 },
      { text: '😆', score: 0.078554883599 },
      { text: '😹', score: 0.063283003867 },
      { text: '😄', score: 0.034623019397 },
      { text: '😝', score: 0.023925462738 },
      { text: '😭', score: 0.019115069881 },
      { text: '😜', score: 0.015078984201 },
      { text: '👇', score: 0.013672497123 },
      { text: '😀', score: 0.012559996918 },
      { text: '🙈', score: 0.011934921145 },
    ],
    synonyms: [],
  });
  await expect(emoji()).resolves.toMatchSnapshot();
});
