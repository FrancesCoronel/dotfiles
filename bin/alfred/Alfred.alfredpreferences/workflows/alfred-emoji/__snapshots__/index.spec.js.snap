// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return empty array 1`] = `
Array [
  Object {
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f440.png",
    },
    "title": "Nothing found!",
  },
]
`;

exports[`should return return correct smiles for "rofl" 1`] = `
Array [
  Object {
    "arg": "😂",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f602.png",
    },
    "subtitle": "face with tears of joy",
    "title": "",
  },
  Object {
    "arg": "😆",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f606.png",
    },
    "subtitle": "smiling face with open mouth and tightly-closed eyes",
    "title": "",
  },
  Object {
    "arg": "😹",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f639.png",
    },
    "subtitle": "cat face with tears of joy",
    "title": "",
  },
  Object {
    "arg": "😄",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f604.png",
    },
    "subtitle": "smiling face with open mouth and smiling eyes",
    "title": "",
  },
  Object {
    "arg": "😝",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f61d.png",
    },
    "subtitle": "face with stuck-out tongue and tightly-closed eyes",
    "title": "",
  },
  Object {
    "arg": "😭",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f62d.png",
    },
    "subtitle": "loudly crying face",
    "title": "",
  },
  Object {
    "arg": "😜",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f61c.png",
    },
    "subtitle": "face with stuck-out tongue and winking eye",
    "title": "",
  },
  Object {
    "arg": "👇",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f447.png",
    },
    "subtitle": "white down pointing backhand index",
    "title": "",
  },
  Object {
    "arg": "😀",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f600.png",
    },
    "subtitle": "grinning face",
    "title": "",
  },
  Object {
    "arg": "🙈",
    "icon": Object {
      "path": "/Users/jukben/Dev/alfred-emoji/node_modules/emoji-datasource-apple/img/apple/64/1f648.png",
    },
    "subtitle": "see-no-evil monkey",
    "title": "",
  },
]
`;
